lib/output.js

```diff
@@ -597,6 +598,12 @@ function OutputStream(options) {
             return true;
         }
 
+        if (output.option('webkit')
+            && output.parent(0) instanceof AST_PropAccess
+            && output.parent(1) instanceof AST_Assign) {
```

Comment:
Also Safari 5 would be happy with `(function(){1}.a=1);` - so you need some form of binary expression in there to trip on this missing parenthesis issue.

But I got lazy with the conditions, because it also trips on things like `(function(){for(;1+1;);}.a=1);` (but not `(function(){for(;;);}.a=1);`) and doing a `TreeWalker` on this seems to be overkill.

Thanks for the reminder about checking `AST_PropAccess.expression` - will fix now.